bftengine/src/bftengine/messages/PartialProofsSet.cpp

```diff
@@ -304,19 +305,12 @@ void PartialProofsSet::tryToCreateFullProof() {
 
 IThresholdVerifier* PartialProofsSet::thresholdVerifier(CommitPath cPath) {
   // TODO: Not thread-safe?
-  IThresholdVerifier* verifier;
-
   // TODO: ALIN: Not sure if the commented code below would be the desired behavior
   if (cPath == CommitPath::OPTIMISTIC_FAST) {
-    verifier = replica->getThresholdVerifierForOptimisticCommit();
+    return CryptoManager::instance().thresholdVerifierForOptimisticCommit();
   } else /* if (cPath == CommitPath::FAST_WITH_THRESHOLD) */ {
-    verifier = replica->getThresholdVerifierForCommit();
-    //} else {
+    return CryptoManager::instance().thresholdVerifierForCommit();
```

Comment:
I might be wrong, but it seems that the class `PartialProofsSet` caches the pointer e.g. `IThresholdAccumulator* thresholdAccumulatorForFast`, might it be a problem in case of rotation?

Reply:
yes, we should remove the usage of
` IThresholdAccumulator* thresholdAccumulatorForFast;`
 ` IThresholdAccumulator* thresholdAccumulatorForOptimisticFast;`